src/storage/block.ts

```diff
@@ -214,8 +263,10 @@ export async function queryBlockCount(): Promise<number> {
 
 export async function queryLatestBlocks(count: number): Promise<DbBlock[]> {
   try {
-    const sql = `SELECT * FROM blocks ORDER BY number DESC LIMIT ${count}`
-    const blocks: DbBlock[] = await db.all(sql)
+    const sql = `SELECT *${config.postgresEnabled ? ', readableBlock::TEXT' : ''} FROM blocks ORDER BY number DESC LIMIT ${count}`
```

Comment:
Instead of something like this can we abstract the query creation into a method (of a class maybe) and have the conditionals in that method instead of having it in place? This is hard to debug in case of issues and from readability POV it's not pleasant. Same applies for the rest of the queries.

Reply:
@arhamj, due to the differences in sql syntax of postgres and sqlite, these queries are different. 
If I understand this correctly, having a separate method that I call to abstract out `select * from ...`, `select field from ...` queries would result in that method becoming a rule-based  mess. Please let me know if this isn't what you had in mind. Also, could you give me an example? That'll help me visualise the solution better.